src/test/ui/pattern/move-ref-patterns/borrowck-move-ref-pattern.stderr

```diff
@@ -189,18 +189,19 @@ LL |     drop(_x3);
 error[E0382]: use of moved value: `tup`
   --> $DIR/borrowck-move-ref-pattern.rs:43:14
    |
-LL |     let mut tup = (U, U, U);
-   |         ------- move occurs because `tup` has type `(U, U, U)`, which does not implement the `Copy` trait
-LL |     let c1 = || {
-   |              -- value moved into closure here
-LL |         let (ref _x0, _x1, _) = tup;
-   |                                 --- variable moved due to use in closure
-LL |     };
-LL |     let c2 = || {
-   |              ^^ value used here after move
-LL |
-LL |         let (ref mut _x0, _, _x2) = tup;
-   |                                     --- use occurs due to use in closure
+LL |       let mut tup = (U, U, U);
+   |           ------- move occurs because `tup` has type `(U, U, U)`, which does not implement the `Copy` trait
+LL |       let c1 = || {
+   |                -- value moved into closure here
+LL |           let (ref _x0, _x1, _) = tup;
+   |                                   --- variable moved due to use in closure
+LL |       };
+LL |       let c2 = || {
```

Comment:
Hmm this is pointing to the wrong message. We might want to add something like this in [this](https://github.com/sexxi-goose/rust/blob/4de57800234bfeea99cb156aa78b52f955491596/compiler/rustc_middle/src/ty/mod.rs#L695-L709) PR and then use it

Reply:
@roxelo This is still needs to be fixed, I should probably spend sometime and see where this is coming from

Reply:
I built this branch with the part that adds the fake reads onto the cfg commented out, and the error was reported as expected.